sources/packages/web/src/services/http/SupportingUserApi.ts

```diff
@@ -13,21 +13,16 @@ export class SupportingUserApi extends HttpBaseClient {
     supportingUserType: SupportingUserType,
     payload: ApplicationIdentifierApiInDTO,
   ): Promise<ApplicationApiOutDTO> {
+    let response;
     try {
-      const response = await this.apiClient.post(
+      response = await this.postCall(
         this.addClientRoot(`supporting-user/${supportingUserType}/application`),
         payload,
-        this.addAuthHeader(),
       );
-      return response.data;
-    } catch (error) {
-      if (!error.response.data?.errorType) {
-        // If it is an not expected error,
-        // handle it the default way.
-        this.handleRequestError(error);
-      }
-      throw error;
+    } catch (error: unknown) {
+      this.handleAPICustomError(error);
     }
+    return response;
```

Comment:
Do we really need a response here? are we doing something with it? If not required not need to return anything.

Also on error the undefined response it returned here which doesn't seem to be right.

Reply:
The response is the application details used in this use case. What should we return in case of error?

Reply:
This is a not-so-usual method that does a POST to execute a "GET" because the information sent can be considered sensitive and would be open if we use the GET instead.

Reply:
I also recommend using Get instead.